pkg/controller/cluster/upgrades.go

```diff
@@ -309,3 +317,64 @@ func (c *Controller) upgrade430(ctx context.Context, tenant *miniov2.Tenant) (*m
 
 	return c.updateTenantSyncVersion(ctx, tenant, version430)
 }
+
+// Upgrades the sync version to v4.4.7
+// in this version we finally deprecated tenant.spec.credsSecret field.
+func (c *Controller) upgrade447(ctx context.Context, tenant *miniov2.Tenant) (*miniov2.Tenant, error) {
+	// If configuration secret is not present build a new one based on the existing credsSecret and delete previous secret after finish
+	if !tenant.HasConfigurationSecret() {
+		if tenant.HasCredsSecret() {
+			credsSecret, err := c.kubeClientSet.CoreV1().Secrets(tenant.Namespace).Get(ctx, tenant.Spec.CredsSecret.Name, metav1.GetOptions{})
+			if err != nil && !k8serrors.IsNotFound(err) {
+				return nil, err
+			}
+			var accessKey string
+			var secretKey string
+
+			if _, ok := credsSecret.Data["accesskey"]; ok {
+				accessKey = string(credsSecret.Data["accesskey"])
+			}
+			if _, ok := credsSecret.Data["secretkey"]; ok {
+				secretKey = string(credsSecret.Data["secretkey"])
+			}
+			if accessKey == "" || secretKey == "" {
+				return nil, fmt.Errorf("'%s/%s' Error migrating tenant credsSecret to Configuration", tenant.Namespace, tenant.Name)
+			}
+			tenantConfiguration := map[string]string{}
+			tenantConfiguration["MINIO_BROWSER"] = "on"
+			tenantConfiguration["MINIO_ROOT_USER"] = accessKey
+			tenantConfiguration["MINIO_ROOT_PASSWORD"] = secretKey
+			configurationSecret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      tenant.ConfigurationSecretName(),
+					Namespace: tenant.Namespace,
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Secret",
+					APIVersion: v1.SchemeGroupVersion.Version,
+				},
+				Data: map[string][]byte{
+					"config.env": []byte(miniov2.GenerateTenantConfigurationFile(tenantConfiguration)),
+				},
+			}
+			_, err = c.kubeClientSet.CoreV1().Secrets(tenant.Namespace).Create(ctx, configurationSecret, metav1.CreateOptions{})
+			if err != nil {
+				return nil, err
+			}
+			// Update tenant fields
+			tenantCopy := tenant.DeepCopy()
+			tenantCopy.EnsureDefaults()
+			tenantCopy.Spec.Configuration = &corev1.LocalObjectReference{
+				Name: tenantCopy.ConfigurationSecretName(),
+			}
+			tenantCopy.Spec.CredsSecret = nil
+			_, err = c.minioClientSet.MinioV2().Tenants(tenant.Namespace).Update(ctx, tenantCopy, metav1.UpdateOptions{})
+			if err != nil {
+				klog.Errorf("Error updating tenant '%s/%s', could not update tenant.spec.configuration field: %v", tenant.Namespace, tenant.Name, err)
+			}
+		} else {
+			return nil, fmt.Errorf("'%s/%s' Error migrating tenant credsSecret, credsSecret not present", tenant.Namespace, tenant.Name)
+		}
+	}
+	return c.updateTenantSyncVersion(ctx, tenant, version447)
+}
```

Comment:
We should perhaps do this for v4.4.8 -made a new release to make sure to fix an important bug during deployment.